File: 524-longest-word-in-dictionary-through-deleting/longest-word-in-dictionary-through-deleting.js
var findLongestWord = function(s, dictionary) {

    dictionary.sort((a, b)=> b.length == a.length ? a.toLowerCase().localeCompare(b.toLowerCase()) : b.length - a.length)
    
    function isSubseq(s, t) {
        let i = 0
        let j = 0

        while (j < t.length) {
            if (s[i] == t[j]) i += 1
            j += 1
        }   

        return i == s.length
    };

    let output = ""

    for (let word of dictionary) {
        if (isSubseq(word, s)) return word
    }
    return output
};

File: 524-longest-word-in-dictionary-through-deleting/longest-word-in-dictionary-through-deleting.py
class Solution:
    def findLongestWord(self, s: str, dictionary: List[str]) -> str:
        dictionary.sort(key = lambda x: (-len(x), x))

        def isSubSeq(part, whole):
            i = 0
            j = 0

            while (j < len(whole) and i < len(part)):
                if (part[i] == whole[j]):
                    i += 1
                j += 1

            return i == len(part)

        output = ""    
        for word in dictionary:
            if (isSubSeq(word, s)):
                return word

        return output

        